lib/rubocop/cop/rspec/implicit_subject.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for usage of implicit subject (`is_expected`).
+      #
+      # This cop can be configured using the `EnforcedStyle` option
+      #
+      # @example `EnforcedStyle: single_line_only`
+      #   # bad
+      #   it do
+      #     is_expected.to be_truthy
+      #   end
+      #
+      #   # good
+      #   it { is_expected.to be_truthy }
+      #   it do
+      #     expect(subject).to be_truthy
+      #   end
+      #
+      # @example `EnforcedStyle: disallow`
+      #   # bad
+      #   it { is_expected.to be_truthy }
+      #
+      #   # good
+      #   it { expect(subject).to be_truthy }
+      #
+      class ImplicitSubject < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = "Don't use implicit subject.".freeze
+
+        def_node_matcher :implicit_subject?, '(send nil? :is_expected)'
```

Comment:
Should we support `:should` and `:should_not` too? Or do we have a policy of not supporting the pre-rspec 3 syntax?

I noticed that `ImplicitExpect` support the older syntax, but I don’t know if any other cop does.

Reply:
Since `should` is available in RSpec 3 (even with the `expect` expectation syntax) and we even have a cop that can be configured to prefer `should` over `is_expected`, it's better to cover them both here.

I'll update later today